src/Libraries/Microsoft.Extensions.AI/Functions/AIFunctionFactory.cs

```diff
@@ -248,9 +245,36 @@ public static ReflectionAIFunctionDescriptor GetOrCreate(MethodInfo method, AIFu
 
         private ReflectionAIFunctionDescriptor(DescriptorKey key, JsonSerializerOptions serializerOptions)
         {
+            // Augment the schema options to exclude AIFunctionArguments/IServiceProvider from the schema,
+            // as it'll be satisfied from AIFunctionArguments.
+            static bool IncludeNonAIFunctionArgumentParameter(ParameterInfo parameterInfo) =>
+                parameterInfo.ParameterType != typeof(AIFunctionArguments) &&
```

Comment:
Curious if we should consider using `IServiceProviderIsService` to filter out other parameter types, similar to how AspNet does it.

Reply:
This would need to be passed into the factory somehow, although I'm not sure there's a clean way to do that.

Reply:
The intent with the current design is that we don't automatically supply parameters from the `IServiceProvider` based on the type, but rather we only supply the `IServiceProvider` itself.

Not certain if that addresses your question though!